lib/msal-core/samples/react-sample-app/src/AuthProvider.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from "react";
+import {
+    msalApp,
+    requiresInteraction,
+    fetchMsGraph,
+    isIE,
+    GRAPH_ENDPOINTS,
+    GRAPH_SCOPES,
+    GRAPH_REQUESTS
+} from "./auth-utils";
+
+// If you support IE, our recommendation is that you sign-in using Redirect APIs
+const useRedirectFlow = isIE();
+
+export default C =>
+    class AuthProvider extends Component {
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                account: null,
+                error: null,
+                emailMessages: null,
+                graphProfile: null
+            };
+        }
+
+        async acquireToken(request, redirect) {
+            return msalApp.acquireTokenSilent(request).catch(error => {
+                // Call acquireTokenPopup (popup window) in case of acquireTokenSilent failure
+                // due to consent or interaction required ONLY
+                if (requiresInteraction(error.errorCode)) {
+                    return redirect
+                        ? msalApp.acquireTokenRedirect(request)
+                        : msalApp.acquireTokenPopup(request);
+                }
+            });
+        }
+
+        async onSignIn(redirect) {
+            if (redirect) {
+                return msalApp.loginRedirect(GRAPH_REQUESTS.LOGIN);
+            }
+
+            const loginResponse = await msalApp
+                .loginPopup(GRAPH_REQUESTS.LOGIN)
+                .catch(() => {
+                    this.setState({
+                        error: "Unable to acquire access token."
+                    });
+                });
+
+            if (loginResponse) {
+                this.setState({
+                    account: loginResponse.account,
+                    error: null
+                });
+
+                const tokenResponse = await this.acquireToken(
+                    GRAPH_REQUESTS.EMAIL
+                );
+
+                const graphProfile = await fetchMsGraph(
+                    GRAPH_ENDPOINTS.ME,
+                    tokenResponse.accessToken
+                );
+
+                this.setState({
+                    graphProfile
+                });
+            }
+        }
+
+        onSignOut() {
+            msalApp.logout();
+        }
+
+        async onRequestEmailToken() {
+            const tokenResponse = await this.acquireToken(
+                GRAPH_REQUESTS.EMAIL,
+                useRedirectFlow
+            ).catch(e => {
+                this.setState({
+                    error: "Unable to acquire access token for reading email."
```

Comment:
Do you want to "reset" error to null on new requests somewherE?

Reply:
Yeah, I reset this in the other places when API calls are made.